src/ts/components/ecomodecontainer.ts

```diff
@@ -0,0 +1,126 @@
+import { SettingsPanelItem } from './settingspanelitem';
+
+export class EcoModeContainer extends Container<ContainerConfig> {
+  private ecoModeSavedEmissionsItem: SettingsPanelItem;
+  private ecoModeToggleButtonItem: SettingsPanelItem;
+  private emissionsSavedLabel: Label<LabelConfig>;
+  private savedEmissons = 0;
+  private currentEnergyEmission: number;
+
+  constructor(config: ContainerConfig = {}) {
+    super(config);
+
+    const ecoModeToggleButton = new EcoModeToggleButton();
+    const labelEcoMode = new Label({
+      text: i18n.getLocalizer('ecoMode.title'),
+      for: ecoModeToggleButton.getConfig().id,
+      id: 'ecomodelabel',
+    });
+    this.emissionsSavedLabel = new Label({
+      text: `${this.savedEmissons.toFixed(4)} gCO2`,
+      cssClass: 'ui-label-savedEnergy',
+    });
+
+    this.ecoModeToggleButtonItem = new SettingsPanelItem(labelEcoMode, ecoModeToggleButton);
+    this.ecoModeSavedEmissionsItem = new SettingsPanelItem('Saved Emissions', this.emissionsSavedLabel, {
+      hidden: true,
+    });
+
+    this.addComponent(this.ecoModeToggleButtonItem);
+    this.addComponent(this.ecoModeSavedEmissionsItem);
+
+    ecoModeToggleButton.onToggleOn.subscribe(() => {
+      this.ecoModeSavedEmissionsItem.show();
+      this.onToggleCallback();
+    });
+
+    ecoModeToggleButton.onToggleOff.subscribe(() => {
+      this.ecoModeSavedEmissionsItem.hide();
+      this.onToggleCallback();
+    });
+  }
+
+  private onToggleCallback: () => void;
+
+  public setOnToggleCallback(callback: () => void) {
+    this.onToggleCallback = callback;
+  }
+
+  configure(player: PlayerAPI): void {
+    player.on(player.exports.PlayerEvent.SegmentPlayback, (segment: SegmentPlaybackEvent) => {
+      if (!segment.mimeType.includes('video')) {
+        return;
+      }
+
+      const { height, width, bitrate, frameRate } = segment.mediaInfo;
+      const {
+        height: maxHeight,
+        bitrate: maxBitrate,
+        width: maxWidth,
+      } = this.getMaxQualityAvailable(player.getAvailableVideoQualities());
+
+      const currentEnergyKwh = this.calculateEnergyConsumption(frameRate, height, width, bitrate, segment.duration);
+
+      const maxEnergyKwh = this.calculateEnergyConsumption(
+        frameRate,
+        maxHeight,
+        maxWidth,
+        maxBitrate,
+        segment.duration,
+      );
+
+      if (this.ecoModeSavedEmissionsItem.isShown()) {
+        this.updateSavedEmissions(currentEnergyKwh, maxEnergyKwh, this.emissionsSavedLabel);
+      } else {
+        this.emissionsSavedLabel.setText(this.savedEmissons.toFixed(4) + ' gCO2');
+      }
+    });
+  }
+
+  updateSavedEmissions(
+    currentEnergyConsuption: number,
+    maxEnergyConsuption: number,
+    emissionsSavedLabel: Label<LabelConfig>,
+  ) {
+    this.currentEnergyEmission = currentEnergyConsuption * 475; // 475 is the average country intensity of all countries in gCO2/kWh
+    const maxEnergyEmisson = maxEnergyConsuption * 475;

```

Comment:
Can we pull out `475` into a variable `carbonIntensity`? 🙏

Reply:
[Move avg carbon intensity to const](https://github.com/bitmovin/bitmovin-player-ui/pull/571/commits/da4f2ff84616282e04adc1f0f128a37ea2253715)